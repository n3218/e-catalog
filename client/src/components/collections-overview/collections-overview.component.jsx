import React from "react"
import { compose } from "redux"
import { connect } from "react-redux"
import { createStructuredSelector } from "reselect"
import { selectCollectionsForPreview, selectIsCollectionFitching } from "../../redux/shop/shop.selectors"
import WithSpinner from "../with-spinner/with-spinner.component"
import CollectionPreview from "../collection-preview/collection-preview.component"
import "./collections-overview.styles.scss"

export const CollectionsOverview = ({ collections }) => {
  return (
    <div className="collection-overview">
      {collections.map(({ id, ...otherProps }) => (
        <CollectionPreview key={id} {...otherProps} />
      ))}
    </div>
  )
}

const mapStateToProps = createStructuredSelector({
  isLoading: selectIsCollectionFitching,
  collections: selectCollectionsForPreview
})

export default compose(connect(mapStateToProps), WithSpinner)(CollectionsOverview)
