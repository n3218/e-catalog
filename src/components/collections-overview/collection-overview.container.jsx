import { compose } from 'redux';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectIsCollectionFitching } from '../../redux/shop/shop.selectors';
import WithSpinner from '../with-spinner/with-spinner.component';
import CollectionsOverview from './collections-overview.component';


const mapStateToProps = createStructuredSelector({
    isLoading: selectIsCollectionFitching
})

const CollectionOverviewContainer = compose(
    connect(mapStateToProps),
    WithSpinner)
    (CollectionsOverview)


export default CollectionOverviewContainer;