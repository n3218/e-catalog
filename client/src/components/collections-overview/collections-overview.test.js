import { shallow, mount, render } from "enzyme"
import React from "react"
import { CollectionsOverview } from "./collections-overview.component"

const mockCollections = [{ id: 1 }, { id: 2 }, { id: 3 }]

it("expect to render CollectionsOverview component", () => {
  expect(shallow(<CollectionsOverview collections={mockCollections} />)).toMatchSnapshot()
})
