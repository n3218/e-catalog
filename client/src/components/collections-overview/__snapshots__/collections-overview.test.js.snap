// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CollectionsOverview component 1`] = `ShallowWrapper {}`;
